App Submission: Agent Zero

slimatic opened the pull request:
# App Submission
### App name
Agent Zero

<img width="256" height="256" alt="a0-icon" src="https://github.com/user-attachments/assets/907eaee0-7f39-4dcb-ba8b-87fd8849b9b0" />

### Gallery images
<img width="1440" height="900" alt="gallery-1" src="https://github.com/user-attachments/assets/7717c776-c9ca-4beb-b9e8-0ec9d8f190e1" />

<img width="1440" height="900" alt="gallery-2" src="https://github.com/user-attachments/assets/562267cb-03f1-4b46-9c7a-f43a48ea55ab" />

<img width="1440" height="900" alt="gallery-3" src="https://github.com/user-attachments/assets/231cbf48-304b-43ce-ad9e-a5ce7651c3aa" />

### Description
Agent Zero is a personal, organic agentic framework that grows and learns with you. It is fully Dockerized and includes features like Speech-to-Text, TTS, and computer use capabilities.
**Key Features:**
- **General Purpose Assistant:** Can handle various tasks.
- **Computer Use:** Can use the computer (in the container).
- **Coding:** Can write and execute code.
- **Memory:** Learns from interactions.
**App Details:**
- **App ID:** `agent-zero`
- **Version:** 0.9.7
- **Port:** 3005
- **Category:** AI
- **Dependencies:** Ollama (recommended for local models)
**Testing:**
### I have tested my app on:
- [ ] umbrelOS on a Raspberry Pi
- [ ] umbrelOS on an Umbrel Home
- [ ] umbrelOS on Linux VM

Commit:
Add Agent Zero app

Commit:
Fix linting errors

Commit:
Use v0.9.7 with multi-arch support (AMD64 and ARM64)

Commit:
Update gallery images and restore gallery in manifest

Commit:
Replace gallery images with new screenshots from local instance

Commit:
Clear gallery in manifest to satisfy linter (images remain in repo)

Commit:
Update gallery image 3 with dashboard screenshot

Commit:
Add Agent Zero icon from GitHub avatar

Commit:
Update Agent Zero icon from local instance

al-lac commented on agent-zero/docker-compose.yml:
```diff
@@ -0,0 +1,18 @@
+    environment:
+      APP_HOST: agent-zero_web_1
+      APP_PORT: 80
+      PROXY_AUTH_ADD: "false"
```
As there is no app authentication, the proxy should be used here.
```suggestion
```

al-lac commented on agent-zero/umbrel-app.yml:
```diff
@@ -0,0 +1,36 @@
+  ⚠️ **Warning:** Agent Zero can execute code. Run with caution.
+developer: Agent Zero AI
+website: https://github.com/agent0ai/agent-zero
+submitter: Agent Zero Community
```
Add yourself as submitter :-) 
```suggestion
submitter: slimatic
```

al-lac commented on agent-zero/gallery/2.png:
Please remove the file from here.

al-lac reviewed (changes requested):
Hey @slimatic,

thanks a lot for this really cool submission! 🤖 
We appreciate the work you put into this. 🤝 

The app has a lot of potential and is a great use case for self hosting.

I added some things that should be addressed before we can get the app into the store.

Using ollama and a small model also leads to endless loops, works way better when using something like gemini. 👍

slimatic committed:
Update agent-zero/docker-compose.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

slimatic committed:
Update agent-zero/umbrel-app.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

slimatic committed:
Update agent-zero/umbrel-app.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

slimatic committed:
Update agent-zero/umbrel-app.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

slimatic committed:
Update agent-zero/docker-compose.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

slimatic committed:
Update agent-zero/docker-compose.yml

Co-authored-by: Alexander L. <lackner.alex@gmail.com>

Commit:
Address PR review feedback

- Add exports.sh script to expose APP_AGENTZERO_PORT and APP_AGENTZERO_LOCAL_URLS
- Remove icon and gallery images (will be handled by Umbrel team)

slimatic commented:
@al-lac checking in to see if I'm missing anything.

slimatic commented:
Hi @al-lac Happy New year. I am just checking in on this. It might have been missed. :-)

al-lac commented:
Hey @slimatic! Happy new year to you as well.

We are still preparing the gallery assets, nothing to do from your side as of now :-) 

We will update you here once the assets are ready.

al-lac committed:
Update to version 1.9

al-lac committed:
Add gallery assets

al-lac commented:
Hey @slimatic, thanks again for your submission of agent-zero. We appreciate the effort you put into this. 🤝 

I updated the version, added the gallery assets and tested the app one more time.

Ready to go live now! 🎉 

<img width="2728" height="1784" alt="CleanShot 2026-04-17 at 14 07 33@2x" src="https://github.com/user-attachments/assets/a17857a1-768b-46bc-88fd-52e499cbf455" />

slimatic commented:
Good to hear! 🎉 Thank you @al-lac !
